### ISSUE TITLE
Question on how to display images of the molecules

### ISSUE THREAD

**Issue by ErikCVik** [rdkit/rdkit#2974]
I should preempt my question by stating I am very new to using rdkit.

I was wondering how to make the images of the files to show after a run. To be clear, I can execute the following code to output the image to an external file.

"""""""""""""""""""""""""""""""""
from rdkit import Chem
from rdkit.Chem import Draw

glycine = Chem.MolFromSmiles('C(C(=O)O)N')
Draw.MolToFile(glycine, 'glycine.png')
"""""""""""""""""""""""""""""""""

However, when I try to execute the following code I do not observe an image.

"""""""""""""""""""""""""""""""""
smiles_list = ['N[C@@H](CC1=CC=CC=C1)C(O)=O', 'C(C(=O)O)N', 'O=C([C@H](CC1=CNC=N1)N)O', 'C([C@@H](C(=O)O)N)S']
mol_list = []
for smiles in smiles_list:
    mol = Chem.MolFromSmiles(smiles)
    mol_list.append(mol)

Draw.MolsToGridImage(mol_list)
"""""""""""""""""""""""""""""""""

To summarize my questions, am I suppose to have an image output after executing the prior command? If so why do you think I do not have one, and if not how do I view the image?

**Comment by greglandrum** [comment]
`Draw.MolsToGridImage()` returns the image data. By default this is a PIL Image object which you can then save. Something like this should work:
```
img = Draw.MolsToGridImage(mol_list)
img.save('output.png')
```

**Comment by ErikCVik** [comment]
You are correct the output is a PIL image, and I was able to save the file as an output to observe what is inside of it by using the code you provided. 

I have another question now, are all Draw objects suppose to be a PIL image? I am now trying to visualize the morgan fingerprints of but the object is a string type rather than an image. Do you know how I can visualize these fingerprints? See the code below.

"""""""""""""""""""""
from rdkit import Chem
from rdkit.Chem import Draw
from rdkit.Chem import Descriptors
from rdkit.Chem import AllChem
from rdkit import DataStructs
import numpy as np

glycine = Chem.MolFromSmiles('C(C(=O)O)N')
Draw.MolToFile(glycine, 'glycine.png')

dic = {}
fp = AllChem.GetMorganFingerprintAsBitVect(glycine, 2, nBits=1024, bitInfo=dic)
fp_arr = np.zeros(1,)
DataStructs.ConvertToNumpyArray(fp, fp_arr)
fp_arr = np.nonzero(fp_arr)
prints = [(glycine, x, dic) for x in fp.GetOnBits()]
img = Draw.DrawMorganBits(prints, molsPerRow=4, legends=[str(x) for x in fp.GetOnBits()])
print(type(img))

**Comment by ErikCVik** [comment]
I think this may have to do with IPython not working properly.

**Comment by Maron-ChingFungChan** [comment]
> `Draw.MolsToGridImage()` returns the image data. By default this is a PIL Image object which you can then save. Something like this should work:
> 
> ```
> img = Draw.MolsToGridImage(mol_list)
> img.save('output.png')
> ```

Hi, I am also new to rdkit and am also trying to save an image.
For some reason, the default format is not PIL for me. Any idea why?

Below is my code:
************************************
from rdkit import Chem
from rdkit.Chem import Draw
from PIL import Image

unique_smiles_freq_non = dict()
suppl_non = Chem.SDMolSupplier('cid2sids-uracil.sdf')

for mol in suppl_non:
    smiles_non = Chem.MolToSmiles(mol,isomericSmiles=True, canonical=False)
    if smiles_non != "":
        unique_smiles_freq_non[ smiles_non ] = unique_smiles_freq_non.get(smiles_non,0) + 1
    
unique_mols_non = [Chem.MolFromSmiles(smile) for smile in unique_smiles_freq_non.keys()]
all_non = Draw.MolsToGridImage(unique_mols_non, molsPerRow=8, subImgSize=(150, 150))

print(type(all_non))
all_non.save("all_non .png")
************************************

This is what I get when I run the code:
************************************
<class 'IPython.core.display.Image'>
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
<ipython-input-3-f8f309abe9e0> in <module>
     15 
     16 print(type(all_non))
---> 17 all_non.save("all_non .png")
     18 
     19 

AttributeError: 'Image' object has no attribute 'save'
************************************

To summarise, I have created a list of molecules (unique_mols_non) and I used the module **Draw.MolsToGridImage** to create one PIL image with all the structures. However, the format isn't PIL and hence I can't save it as PNG.
Thank you:)

**Comment by Maron-ChingFungChan** [comment]
I also have a relevant question if anyone could answer that'd be great. 

Using a PUG-REST request to PubChem.
How do you Retrieve the SIDs of substances whose synonym is "glucose"?

Thanks!!

**Comment by greglandrum** [comment]
@Maron-ChingFungChan you're seeing that behavior because `MolsToGridImage()` behaves differently when you use it in a Jupyter notebook. Here's a gist demonstrating how to get the raw PNG data and save it to a file:
https://gist.github.com/greglandrum/56a80e84676fc3a24250b821d0faac13

**Comment by Lisaazulada5** [comment]
Hi, 

Like this: (I used pycharm)

m = Chem.MolFromSmiles(MurckoScaffold.MurckoScaffoldSmiles('c1c(CCO)ncnc1C(=O)O'))
print(m)
a = Draw.MolToImage(m)
a.show()
